backend/src/dao/TeamEventAttendanceDao.ts

```diff
@@ -10,6 +10,7 @@ async function materializeTeamEventAttendance(
 ): Promise<TeamEventAttendance> {
   return {
     ...dbTeamEventAttendance,
+    status: dbTeamEventAttendance.status as Status,
```

Comment:
Try changing `DBTeamEventAttendance` `status` field to be type `Status`. Then you can get rid of this line entirely, and we can use `Status` type everywhere instead of `string`.